insetmap/insetmap.go

```diff
@@ -90,22 +90,57 @@ func New(db *pgxpool.Pool, config Config, gCSSDir string, gCSSMap CSSMap, gCSSDe
 
 }
 
-// For retrieves the data for the given mdgid to generate a Map, that can
-// be rendered as an SVG
-func (inset *Inset) For(ctx context.Context, mdgid string, cssKey string) (*Map, error) {
-	var (
-		insetmap Map
-		err      error
-	)
-	if inset == nil {
-		return nil, nil
+func (inset *Inset) initLayers(ctx context.Context, imap *Map, ext *geom.Extent) error {
+	for _, lyr := range inset.Layers {
+		var (
+			geobytes []byte
+			class    string
+		)
+
+		sql := replaceTokens(lyr.SQL, ext)
+		var mlayer mapLayer
+		if debug {
+			log.Println("[DEBUG] running sql: ", sql)
+		}
+		rows, err := inset.Query(ctx, sql)
+		if err != nil {
+			return err
+		}
+		for rows.Next() {
+			err = rows.Scan(&class, &geobytes)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "[ERR] geom failed %v: %v\n", lyr.Name, err)
+				rows.Close()
+				return err
+			}
+			g, err := wkb.DecodeBytes(geobytes)
+			if err != nil {
+				var unknownGeo wkb.ErrUnknownGeometryType
+				if errors.As(err, &unknownGeo) {
+					panic(fmt.Sprintf("Unknown geo %v: %v", lyr.Name, unknownGeo))
+				}
+				rows.Close()
```

Comment:
this is usually handled via a `defer rows.Close()` on the returned `rows` in line 105.